scrub_hub_vite/src/components/ConversationListComponent.tsx

```diff
@@ -42,58 +52,70 @@ export default function ConversationListComponent({ conversations, available_use
 			throw 'Bad conversation request';
 		}
 		const data = await response.json();
-		if (isChatProps(data)) {
-			return data;
+		if (isIncomingConversationDetailsData(data)) {
+			return new IncomingConversationDetails(data, await userInfo.user!.privateKey).toProps();
 		} else {
 			alert('error');
 			throw 'Bad conversation data';
 		}
 	}
 
-	async function openConversation(id: number) {
-		const chat = await conversationClick(id.toString());
+	async function openConversation(conversaiton: ConversationProps) {
+		let chat: ChatProps;
+		if (conversaiton.id < 0) {
+			chat = {
+				conversation_id: conversaiton.id,
+				messages: [],
+				participants: conversaiton.participants,
+			}
+		} else
+			chat = await conversationClick(conversaiton.id.toString());
+		console.log(chat.conversation_id);
```

Comment:
What is going on here?

Reply:
Yeah that definitely should have a comment explaining it.

Reply:
(: